### PR TITLE
Feature/main template type support

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Typing JSDocs

**Did you add tests for your changes?**
N/A

**If relevant, link to documentation update:**
N/A

**Summary**
Add's Type support via JSDocs for MainTemplate class. 

**Does this PR introduce a breaking change?**
No

**Other information**
This PR although may include TS nightly change in package.json actually relies on official version bump PR #7115

### CODE REVIEW

**Comment by TheLarkInn** [comment]
@sokra these errors look like legitimate bugs, I only seeing `Module` subclasses inheriting `this._source`, unless its `RawModule` which has `this.sourceStr`. 

Is there something here I'm missing? cc @ooflorent maybe you've come across this

File: lib/MainTemplate.js [hunk 1]
```diff
@@ -296,6 +309,11 @@ module.exports = class MainTemplate extends Tapable {
 		this.requireFn = "__webpack_require__";
 	}
 
+	/**
+	 *
+	 * @param {any} options render manifest options
+	 * @returns {any[]} returns render nanifest
```
Comment (niieani) [inline lib/MainTemplate.js]: @TheLarkInn nanifest 😂

Comment (TheLarkInn) [reply lib/MainTemplate.js]: Lmaooooo nanifest seems cooler though 😂

Comment (TheLarkInn) [reply lib/MainTemplate.js]: Willfix

**Comment by TheLarkInn** [comment]
Rerunning CI with new TypeScript nightly now in master (which this was dependent on)

**Comment by TheLarkInn** [comment]
Merging now that type errors and lint failures are gone, and with @ooflorent approval.

**Pull request merged**